From 7faf8aeff66f21d5f967a1a95177386ec39f5187 Mon Sep 17 00:00:00 2001
Message-Id: <7faf8aeff66f21d5f967a1a95177386ec39f5187.1432314923.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Wed, 13 May 2015 19:08:34 +0530
Subject: [PATCH 1/5] REVERTME: [VPG]: drm/i915: Update pll before DPMS ON

In CHV, a DPMS ON on resume, before any mode set,
results in a blank out on HDMI, due to missing PLL
programming. This patch enables/updates chv PLLs
before enabling crtc during DPMS ON.

This patch is REVERTME because it will not be needed
once we move to our new forklifted kernel.

Change-Id: I6fa95cdc245855f1167eb4d25aefc179ebb230f7
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-25957
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index e42dc6f..545a588 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5851,6 +5851,7 @@ void intel_crtc_control(struct drm_crtc *crtc, bool enable)
 	struct intel_crtc *intel_crtc = to_intel_crtc(crtc);
 	enum intel_display_power_domain domain;
 	unsigned long domains;
+	int ret;
 
 	if (enable) {
 		if (!intel_crtc->active) {
@@ -5859,7 +5860,13 @@ void intel_crtc_control(struct drm_crtc *crtc, bool enable)
 				intel_display_power_get(dev_priv, domain);
 			intel_crtc->enabled_power_domains = domains;
 
-			dev_priv->display.crtc_enable(crtc);
+			/* Enable PLLs before enabling crtc */
+			ret = dev_priv->display.crtc_mode_set(
+					&intel_crtc->base, 0, 0, NULL);
+			if (ret)
+				DRM_ERROR("Enabling PLL for DPMS ON failed\n");
+			else
+				dev_priv->display.crtc_enable(crtc);
 		}
 	} else {
 		if (intel_crtc->active)
-- 
1.7.9.5

